test/records/test_echo_interface.py

```diff
@@ -103,3 +143,80 @@ def test_upload_file_object(self, _):
+
+        with patch("boto3.resource") as mock_resource:
+            mock_bucket = mock_resource.return_value.Bucket.return_value
+
+            mock_bucket.upload_fileobj.side_effect = ClientError(
+                {"Error": {"Code": "AccessDenied", "Message": "Access Denied"}},
+                "upload_fileobj",
+            )
+
+            test_echo = EchoInterface()
+
+            with pytest.raises(EchoS3Error, match="when uploading file at"):
+                test_echo.upload_file_object(BytesIO(), self.test_image_path)
+
+    @patch(
+        "operationsgateway_api.src.config.Config.config.images.echo_url",
+        config_echo_url,
+    )
+    @patch(
+        "operationsgateway_api.src.config.Config.config.images.echo_access_key",
+        config_echo_access_key,
+    )
+    @patch(
+        "operationsgateway_api.src.config.Config.config.images.echo_secret_key",
+        config_echo_secret_key,
+    )
+    @patch(
+        "operationsgateway_api.src.config.Config.config.images.image_bucket_name",
+        config_image_bucket_name,
+    )
+    @patch("boto3.resource")
+    def test_delete_file_object(self, _):
+        test_echo = EchoInterface()
+
+        test_echo.delete_file_object(self.test_image_path)
+        assert test_echo.bucket.Object.return_value.delete.call_count == 1
+
+        with patch(
+            "operationsgateway_api.src.records.echo_interface.log.error",
+        ) as mock_log_error:
+            test_echo = EchoInterface()
+            test_echo.delete_file_object(self.test_image_path)
+
+            mock_log_error.assert_called_once_with(
+                "The object with key %s still exists. Deletion might not "
+                "have been successful.",
+                self.test_image_path,
+            )
+
+        with patch("boto3.resource") as mock_resource:
+            mock_bucket = mock_resource.return_value.Bucket.return_value
+
+            mock_bucket.Object.return_value.delete.side_effect = ClientError(
+                {"Error": {"Code": "AccessDenied", "Message": "Access Denied"}},
+                "delete",
+            )
+
+            test_echo = EchoInterface()
+
+            with pytest.raises(EchoS3Error, match="when deleting file at"):
+                test_echo.delete_file_object(self.test_image_path)
+
+        with patch("boto3.resource") as mock_resource:
+            mock_bucket = mock_resource.return_value.Bucket.return_value
+
+            mock_bucket.Object.return_value.load.side_effect = ClientError(
+                {"Error": {"Code": "AccessDenied", "Message": "Access Denied"}},
+                "delete",
+            )
+
+            test_echo = EchoInterface()
+            with patch(
+                "operationsgateway_api.src.records.echo_interface.log.info",
+            ) as mock_log_info:
+                test_echo.delete_file_object(self.test_image_path)
+
+                assert mock_log_info.call_count == 1
```

Comment:
I set it to check if there was one log call because there was one at the top of the function


its testing that in this piece of code
```
except ClientError as e:
    if e.response["Error"]["Code"] == "404":
        log.info("%s Deletion successful.", image_path)
else:
    log.error(
```

nothing is ran (the response code wan't 404)

I am not sure how I would test it differently